lib/types/application/ApplicationFormatter.js

```diff
@@ -19,12 +26,33 @@ class ApplicationFormatter extends AbstractFormatter {
 				"/": project.resources.configuration.paths.webapp
 			};
 
-			return this.readManifest(project).then(function(manifest) {
-				if (!manifest["sap.app"] || !manifest["sap.app"].id) {
+			return this.readManifest(project).then((manifest) => {
+				// check for a proper sap.app/id in manifest.json to determine namespace
+				const appId = manifest["sap.app"] && manifest["sap.app"].id;
+				if (!appId) {
 					log.warn(`No "sap.app" ID configuration found in manifest of project ${project.metadata.name}`);
 					return;
 				}
-				project.metadata.namespace = manifest["sap.app"].id.replace(/\./g, "/");
+				// check app id for placeholder and try to read the placeholder from the pom.xml
+				const parts = appId.match(/^\$\{(.*)\}$/);
+				if (parts) {
+					log.warn(`No valid "sap.app" ID found in manifest of project ${project.metadata.name}. Reading "${parts[1]}" from pom.xml...`);
```

Comment:
Is this really a warning? Do we want developers to remove such placeholders? I thought it is still valid for many developers to use them.

Reply:
No. Changed to .verbose